### PR TITLE
add adb configuration to hmd-setup.sh

### PR DESCRIPTION
**Pull request by Bennyboiii**

### CODE REVIEW

File: hmd-setup.sh [hunk 1]
```diff
@@ -106,6 +106,30 @@ case $selected in
             wget -q "$wivrn_url/$client_path" -O "$client_file"
             echo --------------------------------
             echo
+            
+            [ "$(adb devices)" == *no\ permissions* ] && {
+                            
+                echo "Configuring adb"
+                [ -f "/etc/udev/rules.d/99-android.rules" ] && {
```
Comment (SaphiraKai) [inline hmd-setup.sh]: you probably want to change this `&&` to a `||` to touch the file if it *doesn't* exist

File: hmd-setup.sh [hunk 2]
```diff
@@ -106,6 +106,30 @@ case $selected in
...
+                    touch "/etc/udev/rules.d/99-android.rules"
```
Comment (SaphiraKai) [inline hmd-setup.sh]: add `$su_cmd` here! you can't create files under `/etc/` as a regular user

File: hmd-setup.sh [hunk 3]
```diff
@@ -106,6 +106,30 @@ case $selected in
...
+                }
+                $su_cmd echo "SUBSYSTEM=='usb', ATTR{idVendor}=='2833', ATTR{idProduct}=='0183', MODE='0666', GROUP='plugdev'" >> /etc/udev/rules.d/99-android.rules
+                $su_cmd udevadm control --reload-rules
+                $su_cmd service udev restart
+                echo "Unplug and replug Quest 2"
+                read -p '(press enter once device has been replugged)'
+                echo
+            }
+            [ "$(adb devices)" == *unauthorized* ] && {
+                echo "Put on your quest and accept the popup to give authorization to install the WiVRn Client"
```
Comment (SaphiraKai) [inline hmd-setup.sh]: update this text to "Put on your Quest and authorize this machine to install the WiVRn client!"

File: hmd-setup.sh [hunk 4]
```diff
@@ -106,6 +106,30 @@ case $selected in
...
+            [ "$(adb devices)" == *no\ permissions* ] && {
+                            
+                echo "Configuring adb"
+                [ -f "/etc/udev/rules.d/99-android.rules" ] && {
+                    touch "/etc/udev/rules.d/99-android.rules"
+                }
+                $su_cmd echo "SUBSYSTEM=='usb', ATTR{idVendor}=='2833', ATTR{idProduct}=='0183', MODE='0666', GROUP='plugdev'" >> /etc/udev/rules.d/99-android.rules
+                $su_cmd udevadm control --reload-rules
+                $su_cmd service udev restart
+                echo "Unplug and replug Quest 2"
```
Comment (SaphiraKai) [inline hmd-setup.sh]: update this text to "Please unplug and replug your Quest!"

File: hmd-setup.sh [hunk 5]
```diff
@@ -106,6 +106,30 @@ case $selected in
...
+                read -p '(press enter once device has been replugged)'
+                echo
+            }
+            [ "$(adb devices)" == *unauthorized* ] && {
+                echo "Put on your quest and accept the popup to give authorization to install the WiVRn Client"
+                read -p '(press enter once popup is accepted)'
```
Comment (SaphiraKai) [inline hmd-setup.sh]: update this text to "(press enter when done)"

**Review by SaphiraKai — CHANGES_REQUESTED** [review]
Sorry for being pedantic with the text, I like consistency!

**Comment by Bennyboiii** [comment]
Udev and prompts fixed! anything else?

**Review by SaphiraKai — APPROVED** [review]
alright, thanks for the work!

**Pull request merged**